### ISSUE TITLE
while_let_on_iterator Suggesting invalid Code With Refutable Pattern

### ISSUE THREAD

**Issue by ironhaven** [rust-lang/rust-clippy#5577]
When using a while let loop to get the results from the [windows function](https://doc.rust-lang.org/std/primitive.slice.html#method.windows) clippy suggests that the while let be replaced with a regular for loop but a regular for loop will not compile because the pattern matches against a slice which has more possibilities.

## Code
```rust
fn main() {
    let list = ["nail", "shoe", "horse"];
    let mut buf = String::new();
    let mut windows = list.windows(2);
    while let Some([first, second]) = windows.next() {
    //for [first, second] in windows {
     buf.push_str(&format!(
            "For want of a {} the {} was lost.\n",
            first, second
        ));
    }
    print!("{}", buf);
}
```

## Clippy Output
```
warning: this loop could be written as a `for` loop
 --> src/lib.rs:4:39
  |
4 |     while let Some([first, second]) = windows.next() {
  |                                       ^^^^^^^^^^^^^^ help: try: `for [first, second] in windows { .. }`
  |
  = note: `#[warn(clippy::while_let_on_iterator)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#while_let_on_iterator
```

## Version
I can replicate with the clippy on the rust playground and my local clippy (clippy 0.0.212 (6651c1b 2020-04-15)
[Playground Link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a8c9a8ae362436db3501e13fdeff39aa)

**Comment by flip1995** [comment]
Fixed in https://github.com/rust-lang/rust-clippy/pull/5559, but has to be deployed to nightly.

**Comment by ironhaven** [comment]
I saw that issue  but i thought it must have not worked for all cased because it still gave the bad change. I should have tested on nightly clippy before reporting. Thanks!
